florist/api/routes/server/job.py

```diff
@@ -45,3 +45,29 @@ async def new_job(request: Request, job: Job = Body(...)) -> Dict[str, Any]:  #
     assert isinstance(created_job, dict)
 
     return created_job
+
+
+@router.get(path="/{status}", response_description="List jobs with the specified status", response_model=List[Job])
+async def list_jobs_with_status(status: JobStatus, request: Request) -> List[Dict[str, Any]]:
+    """
+    List jobs with specified status.
+
+    Fetches list of Job with max length MAX_RECORDS_TO_FETCH.
+
+    :param status: (JobStatus) The status of jobs to query the Job DB for.
+    :param request: (fastapi.Request) the FastAPI request object.
+    :return: (List[Dict[str, Any]]) A list where each entry is a dictionary with the attributes
+        of a Job instance with the specified status.
+    :raises: (HTTPException) status 400 if status is not a member of the JobStatus Enum.
+    """
+    status = jsonable_encoder(status)
+
+    assert isinstance(status, str)
+    if status not in JobStatus._member_names_:
+        msg = f"status is not valid. status: {status}."
```

Comment:
It would be nice to list all valid statuses so the caller would not need to look for them elsewhere, similar to how it's being done here:

https://github.com/VectorInstitute/FLorist/blob/main/florist/api/routes/server/training.py#L70